Common/Extensions.cs

```diff
@@ -146,6 +146,24 @@ public static string ToSHA256(this string data)
             return hash.ToString();
         }
 
+        /// <summary>
+        /// Lazy string to upper implementation.
+        /// Will first verify the string is not already upper and avoid
+        /// the call to <see cref="string.ToUpper()"/> if possible.
+        /// </summary>
+        /// <param name="data">The string to upper</param>
+        /// <returns>The upper string</returns>
+        public static string LazyToUpper(this string data)
```

Comment:
:D Fun!

Reply:
Martin should make a PR to C# and improve System.String.ToUpper(). :-)